src/ILCompiler.Compiler/src/Compiler/Compilation.cs

```diff
@@ -21,11 +21,23 @@ namespace ILCompiler
 {
     public class CompilationOptions
     {
-        public IReadOnlyDictionary<string, string> InputFilePaths;
-        public IReadOnlyDictionary<string, string> ReferenceFilePaths;
+        /// <summary>
+        /// Associate for each simple name, the list of assembly full path names matching the simple name.
+        /// Useful when including multiple assemblies with same name from different locations.
```

Comment:
We could just check if any of the assemblies passed as arguments to ILC have the same base file name and error out early if they do. This is what .NET Native for UWP does. Makes things way less complicated.